### PR TITLE
HPCC-27584 NP selector should be on dynamic thor-eclagent label value

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by jakesmith** [comment]
@richardkchapman - noticed that I'd left a static app label in for the thor-eclagent - it should have been based on the constructed name (otherwise all agents from other thor's could connect).

Please review.
I could reinstate the 'job' label in this PR, and introduce another label (e.g. 'instance')..?
But it would need to be based on the same value (_HPCC_JOBNAME_), and would need to add one an 'instance' to thor-eclagent as well.

**Comment by richardkchapman** [comment]
It feels correct to me to have app as it was, job with jobname, and a new label to use for selection of which pods get the thormaster network policy

**Comment by jakesmith** [comment]
@richardkchapman - reverted app and job label to what they were, introduced new instance label, which in Thor pods is same as job, and in agents is same as app, such that the manager NP ingress selector can select on both the thor job and the thor-eclagent.

File: helm/hpcc/templates/thor.yaml [hunk 1]
```diff
@@ -331,16 +340,15 @@ spec:
...
     matchLabels:
-      run: {{ $commonCtx.eclAgentName | quote }}
-      app: thor-eclagent
+      app: {{ $commonCtx.eclAgentName | quote }}
   template:
     metadata:
       labels:
-        run: {{ $commonCtx.eclAgentName | quote }}
-        app: thor-eclagent
+        app: {{ $commonCtx.eclAgentName | quote }}
```
Comment (richardkchapman) [inline helm/hpcc/templates/thor.yaml:347]: I would think app should be a constant

File: helm/hpcc/templates/thor.yaml [hunk 2]
```diff
@@ -392,14 +400,15 @@ spec:
   replicas: {{ $commonCtx.thorAgentReplicas }}
   selector:
     matchLabels:
-      run: {{ $commonCtx.thorAgentName | quote }}
+      app: {{ $commonCtx.thorAgentName | quote }}
```
Comment (richardkchapman) [inline helm/hpcc/templates/thor.yaml:403]: And here

File: helm/hpcc/templates/thor.yaml [hunk 3]
```diff
@@ -392,14 +400,15 @@ spec:
...
   template:
     metadata:
       labels:
-        run: {{ $commonCtx.thorAgentName | quote }}
+        app: {{ $commonCtx.thorAgentName | quote }}
```
Comment (richardkchapman) [inline helm/hpcc/templates/thor.yaml:407]: And here

**Review by richardkchapman — CHANGES_REQUESTED** [review]
Not sure what app is now used for but it feels like it should be constant.

**Review by richardkchapman — APPROVED** [review]

**Review by ghalliday — APPROVED** [review]
Looks sensible to me

**Comment by richardkchapman** [comment]
@jakesmith Please squash

**Comment by jakesmith** [comment]
@richardkchapman - squashed

**Pull request merged**